### PR TITLE
Feature: Implement complete Header component

### PR DESCRIPTION
**Pull request by chnthkksn**
Here is a summary of the changes I made

- Set up the React application with necessary libraries and files
- Implemented the unstated-next store and MUI ThemeProvider 
- Created Dashboard page and integrated the Header component
- Added a model to add transactions
- Completed the Header component

### CODE REVIEW

File: client/src/components/header/Header.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import Budget from '../../APIs/budget';
+import AddTransaction from './TransactoinModel';
+
+function Navbar() {
+  const store = Store.useContainer();
+
+  const month = store.month.getMonth();
+  const year = store.month.getFullYear();
+
+  const monthName = [
```
Comment (rayanaradha) [inline client/src/components/header/Header.js]: You can add a Constants File to your project and put all constants there.  https://medium.com/@austinpaley32/how-to-add-a-constants-file-to-your-react-project-6ce31c015774

File: client/src/components/header/Header.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+  useEffect(() => {
+    const getBudget = async () => {
+      try {
+        const data = await Budget.get(`${month}${year}`);
+        store.setTransactions(data.budget?.Transactions || []);
+        store.setBudget(data.budget);
+      } catch (error) {}
+    };
+
+    getBudget();
```
Comment (rayanaradha) [inline client/src/components/header/Header.js]: why did you call the same function (getBudget) twice inside the useEffect?

File: client/src/components/header/Header.js (hunk 1 above)
Comment (chnthkksn) [reply client/src/components/header/Header.js]: moved constants to separate file in commit 73e8a85af84ec125088f272f3b35e2c3455cb3d4

File: client/src/components/header/Header.js (hunk 2 above)
Comment (chnthkksn) [reply client/src/components/header/Header.js]: Fixed issues in useEffects in commit d97d6a5f54186a491fb8e2f6b579d8f813ed86e3

**Pull request merged**